File: schemas/question_results.py
from uuid import UUID

from pydantic import BaseModel


class QuestionResultResponse(BaseModel):

    id: UUID

    run_id: UUID

    question: str

    answer: str

    contexts: str

    latency: float

    faithfulness: float

    answer_relevancy: float

    context_precision: float

    context_recall: float

    model_config = {
        "from_attributes": True
    }

File: frontend/src/api/questionResults.ts
import api from "./axios";

export interface QuestionResult {

    id: string;

    question: string;

    answer: string;

    contexts: string;

    latency: number;

    faithfulness: number;

    answer_relevancy: number;

    context_precision: number;

    context_recall: number;

}

export async function getRunResults(
    runId: string,
) {

    const response =
        await api.get<QuestionResult[]>(
            `/runs/${runId}/results`
        );

    return response.data;

}